tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,523 @@
+            else:
+                return False
+
+        except mrack.errors.MrackError:
+            return False
+
+    def _create(self) -> None:
+        if not self.image:
+            self.image = "fedora-latest"
+
+        data: Dict[str, Any] = {
+            'hardware': self.hardware,
+            'name': f'{self.image}-{self.arch}',
+            'os': self.image,
+            'group': 'linux',
+            }
+
+        response = self.api.create(data)
+
+        if response:
+            self.info('guest', 'has been requested', 'green')
+
+        else:
+            raise ProvisionError(
+                f"Failed to create, response: '{response}'.")
+
+        self.guestname = response["id"] if not response["system"] else response["system"]
+        self.info('guestname', self.guestname, 'green')
+
+        with updatable_message(
+                "status", indent_level=self._level()) as progress_message:
+
+            def get_new_state() -> GuestInspectType:
+                response = self.api.inspect()
+
+                if response["status"] == "Aborted":
+                    raise ProvisionError(
+                        f"Failed to create, "
+                        f"unhandled API response '{response['status']}'."
+                        )
+
+                current = cast(GuestInspectType, response)
+                state = current["status"]
+                state_color = GUEST_STATE_COLORS.get(
+                    state, GUEST_STATE_COLOR_DEFAULT
+                    )
+
+                progress_message.update(state, color=state_color)
+
+                if state in {"Error, Aborted", "Cancelled"}:
+                    raise ProvisionError(
+                        'Failed to create, provisioning failed.'
+                        )
+
+                if state == 'Reserved':
+                    return current
+
+                raise tmt.utils.WaitingIncomplete()
+
+            try:
+                guest_info = tmt.utils.wait(
+                    self, get_new_state, datetime.timedelta(
+                        seconds=self.provision_timeout),
+                    tick=self.provision_tick
+                    )
+
+            except tmt.utils.WaitingTimedOutError:
+                response = self.api.delete()
+                raise ProvisionError(
+                    f'Failed to provision in the given amount '
+                    f'of time (--provision-timeout={self.provision_timeout}).'
+                    )
+
+        self.guest = guest_info['system']
+        self.info('address', self.guest, 'green')
+
+    def start(self) -> None:
+        """
+        Start the guest
+
+        Get a new guest instance running. This should include preparing
+        any configuration necessary to get it started. Called after
+        load() is completed so all guest data should be available.
+        """
+
+        if self.guestname is None or self.guest is None:
+            self._create()
+
+    def remove(self) -> None:
```

Comment:
Perhaps we should provide at least a simple `stop` method as well and just log that for `beaker` no action is done? Otherwise a `NotImplementedError` would be raised when `stop()` is called.